src/OAuth2/GrantType/RefreshToken.php

```diff
@@ -34,6 +34,14 @@ public function __construct(RefreshTokenInterface $storage, $config = array())
             'always_issue_new_refresh_token' => false,
             'unset_refresh_token_after_use' => true
         ), $config);
+
+        // to preserve B.C. with v1.6
+        // @see https://github.com/bshaffer/oauth2-server-php/pull/580
+        // @todo - remove in v2.0
+        if (isset($config['always_issue_new_refresh_token']) && !isset($config['unset_refresh_token_after_use'])) {
+            $this->config['unset_refresh_token_after_use'] = $config['always_issue_new_refresh_token'];
```

Comment:
It should probably be fixed, yes, as this inadvertently changed the default
behavior.
On Thu, Jun 11, 2015 at 12:30 PM Vytautas Stankus <notifications@github.com>
wrote:

> In src/OAuth2/GrantType/RefreshToken.php
> <https://github.com/bshaffer/oauth2-server-php/pull/587#discussion_r32208013>
> :
>
> > @@ -34,6 +34,14 @@ public function __construct(RefreshTokenInterface $storage, $config = array())
> >              'always_issue_new_refresh_token' => false,
> >              'unset_refresh_token_after_use' => true
> >          ), $config);
> > +
> > +        // to preserve B.C. with v1.6
> > +        // @see https://github.com/bshaffer/oauth2-server-php/pull/580
> > +        // @todo - remove in v2.0
> > +        if (isset($config['always_issue_new_refresh_token']) && !isset($config['unset_refresh_token_after_use'])) {
> > +            $this->config['unset_refresh_token_after_use'] = $config['always_issue_new_refresh_token'];
>
> @bshaffer <https://github.com/bshaffer> do you think this should be
> fixed? or this works as expected?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bshaffer/oauth2-server-php/pull/587/files#r32208013>.
>